### ISSUE TITLE
Use www-data user instead of magento2 user

### ISSUE THREAD

**Issue by ihor-sviziev** [magento/magento2devbox-web#30]
Is there any reason why you used own magento2 user instead of "www-data"?

**Comment by ashsmith** [comment]
Great suggestions here 👍  but these should be split up into separate issue and provide more information to help guide Magento, and also useful for those working on these suggestions to have an issue per suggestion.

I've already written up the xdebug suggestion as that'll be great, and the fact it's enabled by default is troublesome.

**Comment by Stolr** [comment]
+ 1 on theses issues. 

The major thing is the splitting. (6.).  You started the job by offering a docker compose to create multiple containers but you definetly should go deeper.  

This docker box can be used to develop a magento2 website but won't be used. Why ?  You can't match the production environment with this configuration. What if your production server runs under nginx webserver ? Using php 7.0.8 ?  

And whaooo no linux support O_O

But thanks for your job, this is great. Keep going to make it better for developer  !

**Comment by piotrekkaminski** [comment]
> Using magento2 user instead of www-data user. Is there any reason why you used own magento2 user instead of "www-data"?

There is no standard user - some sites use www-data, some www, some apache. We wanted to avoid conflicts with existing users.

> Using custom http port. Production websites not using custom ports, as result we can see different issues on production & dev env.

Wouldn't using custom port actually possibly expose more issues than using default? Adding proxy makes sense but adds complexity.

> Https is not used. Production websites must use https at least for checkout, so dev. env. should also have it (i suggest to use self-signed ssl cert)

Good point. Definitely a place for improvement, eg possible integration with Let's encrypt.

> Using specific bash image versions. As example in docker-compose.yml file we have "redis:3.0.7", it should be replaced with "redis:3.0" to use latest available 3.0.x version, similar issue in Dockerfile "FROM php:7.0.12-fpm"

While it makes sense, we want to make sure to test specific version - a lot of variables couple with Docker instability makes it hard to test.

> Using multiple magento installations in parallel is hard...

Interesting idea but adds complexity to add proxy. Something to consider.

> Using "all-in-one" container.

There are some reasons listed in readme. Mostly we had lots of instability issues with docker with more containers.

> SSH and xDebug

Some changes should be coming here. 

> Linux support

We hear you. In backlog.

**Comment by ihor-sviziev** [comment]
@piotrekkaminski 
1. PHP fpm use www-data, Apache - apache user. I think should be used one of these users in order be as much closer to production infra
2. I had issues in our projects that were caused by using custom ports. As for me - I can't use your default configuration because of this issue.
3. I don't think that integration with let's encrypt could be used on dev machine as easy way
4. could you explain which instability issues did you experienced? I'm using very close to production docker images on my local machine, and I haven't any issues. Reason maybe that I'm using linux

**Comment by piotrekkaminski** [comment]
4) containers locked, networking down, unkillable containers (requiring actually a computer reboot) on Mac and just random connectivity issues in Win.

**Comment by ihor-sviziev** [comment]
@piotrekkaminski I think these issues should be reported to https://github.com/docker/docker/issues

**Comment by vrann** [comment]
@ihor-sviziev 
1. You can edit the docker-compose.yml file after downloading and write the port you'd like to use with the services, then run m2devbox-init.sh. Will it help with your issue? By default, we would like to keep unique ports per instance to allow multiple projects running on different compose clusters at the same time.
7. Regarding xdebug let's keep the discussion here https://github.com/magento/magento2devbox-web/issues/32 Haven't seen a viable solution so far and will be glad to see how did you manage it working with the IDE.

**Comment by jigneshthummar** [comment]
@vrann do we need any port forwarding to access it on host machine ? if yes  can you share steps ?
